website/static/js/pages/project-settings-page.js

```diff
@@ -13,28 +13,48 @@ require('css/addonsettings.css');
 
 var ctx = window.contextVars;
 
-
-// Initialize treebeard grid for notifications
 var ProjectNotifications = require('js/notificationsTreebeard.js');
 var $notificationsMsg = $('#configureNotificationsMessage');
 var notificationsURL = ctx.node.urls.api  + 'subscriptions/';
-// Need check because notifications settings don't exist on registration's settings page
-if ($('#grid').length) {
+
+function initNotificationsTB() {
+    // Initialize treebeard grid for notifications
+    // Need check because notifications settings don't exist on registration's settings page
+    if ($('#grid').length) {
+        $.ajax({
+            url: notificationsURL,
+            type: 'GET',
+            dataType: 'json'
+        }).done(function(response) {
+            new ProjectNotifications(response);
+        }).fail(function(xhr, status, error) {
+            $notificationsMsg.addClass('text-danger');
+            $notificationsMsg.text('Could not retrieve notification settings.');
+            Raven.captureMessage('Could not GET notification settings.', {
+                url: notificationsURL, status: status, error: error
+            });
+        });
+    }
+}
+initNotificationsTB();
+
+//Initialize treebeard grid for node admins
+var ProjectMailingList = require('../mailingListTreebeard.js');
+var mailingListSettingsURL = ctx.node.urls.api + 'mailing_list/';
```

Comment:
TODOs are a fallback option, because they assume the solution is to hard code a different URL (same bat-issue, different bat-time)- the best choice would be to pass in the URL along with any other data used to initialize the object.

That said, if the rest of the PR is ready this evening I'll try to wrap my part of the review then, and trust the person merging to make a decision on whether other pending changes to celery are blockers.